llvm/lib/Transforms/Utils/FunctionComparator.cpp

```diff
@@ -804,6 +804,10 @@ int FunctionComparator::cmpOperations(const Instruction *L,
         return Res;
     }
   }
+  if (const CastInst *Cast = dyn_cast<CastInst>(L)) {
+    const CastInst *CastR = cast<CastInst>(R);
+    return cmpTypes(Cast->getDestTy(), CastR->getDestTy());
```

Comment:
@SamTebbs33 do we need a separate test for the FunctionComparator change? (I don't think that's currently tested?)